tests/test_serializer_nested.py

```diff
@@ -302,3 +305,55 @@ class Meta:
             'serializer `tests.test_serializer_nested.DottedAddressSerializer`, '
             'or set `read_only=True` on dotted-source serializer fields.'
         )
+
+
+@pytest.mark.skipif('not postgres_fields')
+class TestNestedNonRelationalFieldWrite:
+    """
+    Test that raise_errors_on_nested_writes does not raise `AssertionError` when the
+    model field is not a relation.
+    """
+
+    def test_nested_serializer_create_and_update(self):
+        class NonRelationalPersonModel(models.Model):
+            """Model declaring a postgres JSONField"""
+            data = postgres_fields.JSONField()
+
+        class NonRelationalPersonDataSerializer(serializers.Serializer):
+            occupation = serializers.CharField()
+
+        class NonRelationalPersonSerializer(serializers.ModelSerializer):
+            data = NonRelationalPersonDataSerializer()
+
+            class Meta:
+                model = NonRelationalPersonModel
+                fields = ['data']
+
+        serializer = NonRelationalPersonSerializer(data={'data': {'occupation': 'developer'}})
+        assert serializer.is_valid()
+        assert serializer.validated_data == {'data': {'occupation': 'developer'}}
+        ModelClass = serializer.Meta.model
+        info = model_meta.get_field_info(ModelClass)
+        raise_errors_on_nested_writes('create', serializer, serializer.validated_data, info)
+        raise_errors_on_nested_writes('update', serializer, serializer.validated_data, info)
+
+    def test_dotted_source_field_create_and_update(self):
+        class NonRelationalPersonModel(models.Model):
+            """Model declaring a postgres JSONField"""
+            data = postgres_fields.JSONField()
+
+        class DottedNonRelationalPersonSerializer(serializers.ModelSerializer):
+            occupation = serializers.CharField(source='data.occupation')
+
+            class Meta:
+                model = NonRelationalPersonModel
+                fields = ['occupation']
+
+        serializer = DottedNonRelationalPersonSerializer(data={'occupation': 'developer'})
+        assert serializer.is_valid()
+        assert serializer.validated_data == {'data': {'occupation': 'developer'}}
+        ModelClass = serializer.Meta.model
+        info = model_meta.get_field_info(ModelClass)
+        raise_errors_on_nested_writes('create', serializer, serializer.validated_data, info)
+        raise_errors_on_nested_writes('update', serializer, serializer.validated_data, info)
+        assert serializer.data == {'occupation': 'developer'}
```

Comment:
out of curiosity, is there a reason to assert the `data` here?

Reply:
You are correct, there is no point asserting this.